define/types.go

```diff
@@ -191,16 +202,73 @@ func cloneToDirectory(url, dir string) ([]byte, string, error) {
 		}
 		gitBranch = gitSubDirPart[0]
 	}
-	if gitBranch == "" {
-		logrus.Debugf("cloning %q to %q", gitBranchPart[0], dir)
-		cmd = exec.Command("git", "clone", "--recurse-submodules", gitBranchPart[0], dir)
-	} else {
-		logrus.Debugf("cloning repo %q and branch %q to %q", gitBranchPart[0], gitBranch, dir)
-		cmd = exec.Command("git", "clone", "--recurse-submodules", "-b", gitBranch, gitBranchPart[0], dir)
+	// Attempt parsing commit, accepted syntax listed below:
+	// * repourl.git#branch:subdir@commit
+	// * repourl.git#branch@commit ( uses default subdir )
+	// * repourl.git@commit ( uses default branch )
```

Comment:
Commit IDs can be referred to independently of branches.  We don't need to add "@" to the syntax.

Reply:
@nalind Yes in previous approach since we were doing `"git", "clone", "--recurse-submodules", "-b"` so branch was always needed but its not needed anymore, amended the PR.